lib/oban.ex

```diff
@@ -203,23 +203,32 @@ defmodule Oban do
 
       Ecto.Multi.new()
       |> Oban.insert("job-1", MyApp.Worker.new(%{id: 1}))
-      |> Oban.insert("job-2", MyApp.Worker.new(%{id: 2}))
+      |> Oban.insert("job-2", fn _ -> MyApp.Worker.new(%{id: 2}) end)
       |> MyApp.Repo.transaction()
   """
   @doc since: "0.7.0"
   @spec insert(
           name :: atom(),
           multi :: Multi.t(),
           multi_name :: Multi.name(),
-          changeset :: Changeset.t(Job.t())
+          changeset_or_fun :: Changeset.t(Job.t()) | fun()
```

Comment:
I think we can parameterize the `fun`, though the spec for `Ecto.Multi.insert/4` looks wrong to me. The function receives a map of changes, not a changeset or schema.